hydrus/resources.py

```diff
@@ -154,8 +156,16 @@ def post(self, id_: str, path: str) -> Response:
                         type_=object_["@type"],
                         session=get_session(),
                         api_name=get_api_name())
-                    headers_ = [{"Location": "{}{}/{}/{}".format(
-                            get_hydrus_server_url(), get_api_name(), path, object_id)}]
+                    method = "POST"
+                    resource_url = "{}{}/{}/{}".format(
+                            get_hydrus_server_url(), get_api_name(), path, object_id)
+                    last_job_id = crud.get_last_modification_job_id(session=get_session())
+                    new_job_id = crud.insert_modification_record(method, resource_url,
+                                                                 session=get_session())
+                    send_sync_update(socketio=socketio, new_job_id=new_job_id,
+                                     last_job_id=last_job_id, method=method,
+                                     resource_url=resource_url)
+                    headers_ = [{"Location": resource_url}]
                     status_description = "Object with ID {} successfully updated".format(object_id)
                     status = HydraStatus(code=200, title="Object updated", desc=status_description)
```

Comment:
I don't know whether client caches collections or not, nevertheless I made a commit adding sync update for PUT operation over collections.

Reply:
I thought a lot about this too and think it's an important architectural discussion I wanted to have.

At first, I also wanted to add PUT to sync however we're using the Agent to cache resources, in that sense, syncing PUT requests would be using Redis more in a "db mirror/copy way" more than caching. I think this architectural decision of syncing PUT requests would be really important if we didn't have IRI Template to search the server.

I think both approaches have different benefits and it depends on the user use case. When imagining a use case where the Server has millions of resources and the Agent is a regular application that, for example, fetches the server with a 5-sized-pagination-size, always copying the whole database then syncing PUT will generate a huge amount of requests which won't be useful and ever used by the user. However, only caching the resources he actually queries for will produce 1 request for each and them save a lot of future requests for that resource since it's cached.

>  "check if there is an item in the collection with name xyz" then what will the agent do

When fetching a hydrus collection it returns a pagination-sized list of "@id" and "@type", so it doesn't have the members details to already cache these resources. So If a user queries "check if there is an item in the collection with name xyz", by default, what will happen is that the Agent will forward the IRI Template that will search the DB internally and return the collection with a list having "@id" and "@type" for the found resources. This list can be used to fetch the resources itself using their ID, which might be already cached or will be cached if not. In that scope, **I still wanted to have a way to cache Search Queries** and not go for the server always.. And for that, I came up with the cached_limit optional parameter so the user can inform how many items he wants to fetch and, if internally Redis has enough resources for that search, it won't go for the server, it will return from the already cached resources.

Give me your feedback on that

Reply:
we are not really interested il storing collections in redis as we should have already all their elements as nodes in the graph. so i would focus in keeping instances updated and just assume that the collection is the output of querying all the nodes of a given type